franklin/core/helpers.py

```diff
@@ -68,3 +76,41 @@ def has_permission(self, request, view):
             is_valid_github = hmac.compare_digest(computed_secret, secret)
             return is_valid_github
         return False
+
+
+class SocialAuthentication(BaseAuthentication):
+    """
+    Used in DEFAULT_AUTHENTICATION_CLASSES settings for authentication.
+    This setting allows users to authenticate with only a Github oauth token.
+    """
+    def authenticate(self, request):
+        auth_header = get_authorization_header(request)\
+                            .decode(HTTP_HEADER_ENCODING)
+        auth = auth_header.split()
+        if not auth or auth[0].lower() != 'bearer':
+            return None
+        if len(auth) == 1:
+            msg = 'Credentials are malformed'
+            raise exceptions.AuthenticationFailed(msg)
+        oauth_token = auth[1]
+
+        social_user = UserSocialAuth.objects\
+                                    .filter(extra_data__contains=oauth_token)\
+                                    .first()
+        if not social_user:
+            # User does not exist in our DB, attempt social auth
+            strategy = load_strategy(request=request)
+            path = NAMESPACE + ":complete"
+            backend = 'github'
+            backend = load_backend(strategy, backend,
+                                   reverse(path, args=(backend,)))
+            user = backend.do_auth(access_token=oauth_token)
+            if not user:
+                msg = 'Bad credentials'
+                raise exceptions.AuthenticationFailed(msg)
+            social = user.social_auth.get(provider='github')
+            social.extra_data['access_token'] = oauth_token
+            social.save()
```

Comment:
Most of this method was borrowed from best practices elsewhere. To summarize: If we don't find the user in the DB, we attempt to `do_auth` with python social auth. If that step succeeds, all the models and everything (with exception to the oauth_token bug) will be saved to the DB for us. Thus, the first time the dashboard calls ANY api endpoint with a users token, we will taken a moment here to initialize that user. All future calls will just hit the DB. New oauth tokens will result in a call to github, but will then just be associated with the user.